pkg-manager/resolve-dependencies/src/resolvePeers.ts

```diff
@@ -78,16 +81,30 @@ export function resolvePeers<T extends PartialResolvedPackage> (
   const rootPkgsByName = opts.resolvePeersFromWorkspaceRoot ? getRootPkgsByName(opts.dependenciesTree, opts.projects) : {}
   const peerDependencyIssuesByProjects: PeerDependencyIssuesByProjects = {}
 
-  for (const { directNodeIdsByAlias, topParents, rootDir, id } of opts.projects) {
+  for (const { directNodeIdsByAlias, manifest, topParents, rootDir, id } of opts.projects) {
     const peerDependencyIssues: Pick<PeerDependencyIssues, 'bad' | 'missing'> = { bad: {}, missing: {} }
     const pkgsByName = {
       ...rootPkgsByName,
       ..._createPkgsByName({ directNodeIdsByAlias, topParents }),
     }
 
+    const npmAliases: Record<string, string[]> = {}
+    const allDependencies = {
+      ...manifest.dependencies,
+      ...manifest.devDependencies,
+    }
+    for (const [pkgName, version] of Object.entries(allDependencies)) {
+      if (version.startsWith('npm:')) {
+        const aliasedPkgName = version.substring(4).split('@')[0]
```

Comment:
I don't think there's a need to parse this info. The package name and alias name are already present in the nodes of dependenciesGraph

Reply:
Although we can get the values in `dependenciesGraph`, there're still two problems:

- `dependenciesGraph` is an object, which means we have to loop through its key to find all dependencies starting with `peerName`, which destroys the performance.
- Even if the performance slowdown is acceptable, there is no alias info in the value in `dependenciesGraph` (the field `name` and `version` look as if it's not aliased).